Userland/Services/LookupServer/LookupServer.cpp

```diff
@@ -77,24 +79,35 @@ LookupServer::LookupServer()
 
 void LookupServer::load_etc_hosts()
 {
-    m_etc_hosts.clear();
-    auto add_answer = [this](Name const& name, RecordType record_type, DeprecatedString data) {
-        m_etc_hosts.ensure(name).empend(name, record_type, RecordClass::IN, s_static_ttl, move(data), false);
+    auto new_hosts_or_error = this->try_load_etc_hosts();
+    if (new_hosts_or_error.is_error())
+        dbgln("Ignoring '/etc/hosts', keeping old values");
+    else
+        m_etc_hosts = new_hosts_or_error.release_value();
+}
+
+ErrorOr<HashMap<Name, Vector<Answer>, Name::Traits>> LookupServer::try_load_etc_hosts()
+{
+    HashMap<Name, Vector<Answer>, Name::Traits> map;
+    auto add_answer = [&map](Name const& name, RecordType record_type, DeprecatedString data) {
+        return map.ensure(name).try_empend(name, record_type, RecordClass::IN, s_static_ttl, move(data), false);
```

Comment:
The resolution to <https://github.com/SerenityOS/serenity/pull/18759#discussion_r1192111603> seems to have been lost. This should use try_ensure.

Reply:
Thanks! It wasn't lost, I just focused too much on the `empend` call, and somehow overlooked the `ensure` call in the same line. Will fix!